src/state/actions/RemoteFeatureAction.ts

```diff
@@ -139,32 +139,39 @@ export function fetchFeatures(type: string, urls: string[], passThroughOpts: any
 
     // Iterate all given URLs and create a dispatcher method for each one.
     urls.forEach((url) => {
+      // Skip internal URLs (temporary layers).
+      if (url.startsWith('internal://')) {
+        return;
+      }
       // eslint-disable-next-line
       dispatcher.push(dispatch(fetchFeaturesFromResource(url, fetchOpts, format, readerOpts)));
     });
 
     // Set loading to true by dispatching the following action.
     dispatch(fetchingFeatures(type, controller, passThroughOpts));
 
-    // Load all single resources.
-    Promise.all(dispatcher)
-      .then(features => {
-        // only dispatch if we're not out of date
-        if (last === currentTime) {
-          const resultFeatures = {};
-          features.forEach(feat => {
-            if (!isEmpty(feat.features)) {
-              Object.assign(resultFeatures, feat.features);
-            }
-          });
-
-          // As soon as all single actions are fulfilled, write the combined
-          // features into the state.
-          return dispatch(fetchedFeatures(type, resultFeatures, passThroughOpts));
-        }
-      }).catch(() => {
-        return dispatch(fetchedFeatures(type, {}, passThroughOpts));
-      });
+    try {
+      // Load all single resources.
+      const dispatchFeatures = await Promise.all(dispatcher);
+
+      // only dispatch if we're not out of date
+      if (last === currentTime) {
+        let resultFeatures = {};
+        dispatchFeatures.forEach(feat => {
+          if (!isEmpty(feat.features)) {
+            Object.assign(resultFeatures, feat.features);
+          }
+        });
+        resultFeatures = {...resultFeatures,...passThroughOpts.internalVectorFeatures};
+
+        // As soon as all single actions are fulfilled, write the combined
+        // features into the state.
+        return dispatch(fetchedFeatures(type, resultFeatures, passThroughOpts));
```

Comment:
Did you check if we could remove the `internalVectorFeatures` from the `passThroughOpts`?

Reply:
please see #799